src/core/utils/modal.tsx

```diff
@@ -52,19 +53,33 @@ function Modal({
     return null;
   }
 
+  const close = () => {
+    dispatch(closeModal({ modalId }));
+  };
+
   return (
     <FocusTrap>
-      <div
-        className="modal-backdrop"
-        // TODO: Close the modal when clicking backdrop.
-        style={{
-          // some elements are designed with z-index which means they pop up over the modal
-          // so I add 10 to the z-index of the modal
-          // the index of the modalid is used, so the newest modal is always on top of
-          // the remaining modals
-          zIndex: modalIds.indexOf(modalId) + 10
-        }}
-      >
+      <div>
+        {/* The backdrop doesn't have a tab index or keyboard listener because it barely duplicates
+          the close button's functionality which possesses both. */}
+        {/* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/interactive-supports-focus */}
+        <div
+          className="modal-backdrop"
+          style={{
+            // some elements are designed with z-index which means they pop up over the modal
+            // so I add 10 to the z-index of the modal
+            // the index of the modalid is used, so the newest modal is always on top of
+            // the remaining modals
+            zIndex: modalIds.indexOf(modalId) + 10
+          }}
+          role="button"
+          onClick={() => {
+            close();
```

Comment:
Consider the usage of the role button.

I appreciate the commentary regarding disabling the accessibility linting issues but I wonder if you got this 100% right with the `role="button"` for the backdrop when we at the same time want assistive technology to ignore the element entirely and defer to the actual close button.

I took a look at how [the Reach UI project (accessible React UI elements) handles dialogs](https://reach.tech/dialog). They do not seem to hav any accessibility-related props on their overlay but still register a click handler to close the dialog when clicking on it.

I expect removing the `button` role might affect which linting errors we get but then I suggest that we adjust these accordingly.

If we want to investigate this we could also try to use some assistive technology ourselves to see how it interprets the current situation.

Reply:
Yeah, it's a bit tricky. And you're totally right about the role button. I removed it from the backdrop, but then we just get different errors (as you mention you'd expect).